src/python/grpcio/grpc/experimental/aio/_server.py

```diff
@@ -13,34 +13,47 @@
 # limitations under the License.
 """Server-side implementation of gRPC Asyncio Python."""
 
-from typing import Text, Optional
 import asyncio
+from concurrent.futures import Executor
+from typing import Any, Optional, Sequence, Text
+
 import grpc
-from grpc import _common
+from grpc import _common, _compression
 from grpc._cython import cygrpc
 
+from ._typing import ChannelArgumentType
+
+
+def _augment_channel_arguments(base_options: ChannelArgumentType,
+                               compression: Optional[grpc.Compression]):
+    compression_option = _compression.create_channel_option(compression)
+    return tuple(base_options) + compression_option
+
 
 class Server:
     """Serves RPCs."""
 
-    def __init__(self, thread_pool, generic_handlers, interceptors, options,
-                 maximum_concurrent_rpcs, compression):
+    def __init__(self, thread_pool: Optional[Executor],
+                 generic_handlers: Optional[Sequence[grpc.GenericRpcHandler]],
+                 interceptors: Optional[Sequence[Any]],
+                 options: ChannelArgumentType,
+                 maximum_concurrent_rpcs: Optional[int],
+                 compression: Optional[grpc.Compression]):
         self._loop = asyncio.get_event_loop()
-        self._server = cygrpc.AioServer(self._loop, thread_pool,
-                                        generic_handlers, interceptors, options,
-                                        maximum_concurrent_rpcs, compression)
+        self._server = cygrpc.AioServer(
+            self._loop, thread_pool, generic_handlers, interceptors,
+            _augment_channel_arguments(options, compression),
```

Comment:
@pfreixes Injected compression channel argument.

Reply:
So in case of having it enabled by default, we do not need to explicitly send [1] any flag or so for saying that compression is enabled? The Core will take care of that, by for example expanding the metadata operations and adding the compression "headers"?

[1] https://github.com/grpc/grpc/pull/21809/files#diff-b53c077f7911f53dc8ce7656e16d35cdR24

Reply:
Yes. If we enable certain algorithm by default, we don't need to call out `set_compression` explicitly.